botocore/client.py

```diff
@@ -491,7 +492,17 @@ def _make_api_call(self, operation_name, api_params):
         )
 
         if http.status_code >= 300:
-            raise ClientError(parsed_response, operation_name)
+            if parsed_response.get("Error", {}).get("Code"):
+                error_code = parsed_response["Error"]["Code"]
+                if "." in error_code:
+                    error_code = error_code.replace(".", "")
+                if error_code.isdigit():
+                    error_class = ClientError
+                else:
+                    error_class = getattr(self.meta.exceptions, error_code)
```

Comment:
@jamesls, I've added the reverse map and error retrieval by code where the model contains that information. Please take a look.